### ISSUE TITLE
some kind of `wrap` function for PBC's?

### ISSUE THREAD

**Issue by rkurchin** [JuliaMolSim/AtomsBase.jl#9]
Would it make sense to have this be part of the interface? If so, we would need to be able to dispatch on boundary condition types, which might require rejiggering the type parameters of `AbstractSystem`...

**Comment by eahenle** [comment]
From my (admittedly biased) perspective, periodic boundaries are very important.  Can't handle crystalline materials without them.  Molecular dynamics simulations usually use them, too.

The math is, as you say, not complicated.  In `Xtals` we keep positions in fractional coordinates so the math simple: anything less than 0 gets 1 added to it, and anything greater than 1 gets 1 taken away.  (+1.5, 0.5, -1) maps to (0.5, 0.5, 0).

**Comment by Gregstrq** [comment]
>I think if you have PBC and you add an absolute coordinate that is "larger than the border" it should always wrap. That is the point as @eahenle has described it. If you don't want that you first "disable" periodicity and then add the position.

Technically, I imagine certain subtleties associated with fast implementations. For example, it may be better first to update all the positions and then to wrap them all back inside the boundary. @jgreener64, how is it usually done in MD simulations?

**Comment by rkurchin** [comment]
These are all fair points. I really opened this issue as a "thought that popped into my head" and have no strong feelings about whether it should be implemented here or not. Certainly given that we're not doing setters, at the most if we had it it would just return the wrapped coordinates rather than modifying them, but at that point I'm not sure there's all that much use to having it in the interface...

**Comment by cortner** [comment]
It feels all these methods to manipulate systems could become part of `AtomsBuilder`

**Comment by rkurchin** [comment]
Agreed – I'm fine with this being closed and having an issue over there instead as a reminder

**Comment by cortner** [comment]
I'm planning to move over the rest of JuLIP functinality to AtomsBuilder over the coming week. If you file an issue there and describe how you think `wrap` should work then I'll try to match that.
